src/classes/CW_AudienceCSVProcessHelper.cls

```diff
@@ -0,0 +1,406 @@
+/**
+* File Name :   CW_AudienceCSVProcessHelper.cls
+* Description : Helper Class for processing iported CSV file in batches
+*               
+* Modification Log
+* ============================================================================
+* Ver Date       Author         Modification
+* --- ---------- -------------- --------------------------
+* 0.1 20/05/2014 Alex PAnka     Initial version
+*
+*
+*/
+public with sharing class CW_AudienceCSVProcessHelper {
+	
+	private Id CampaignID ;
+	private CW_CSV_Import_Status__c CsvStatus;
+	private List<CW_CSV_Stage_AudienceMembers__c> CsvStageList;
+	private Map<ID,CW_Campaign_Reward__c> campRewardMap;  //Reference by Reward Library
+	private Map<Id,CW_Audience_Group_Schedule__c> grupSheduleMap; //Reference by group ID.
+	private map<string,id> audSchedRewMap;//Reference by group ID+Library_id
+	private map<string,id> audMemberMap;//Reference by group ID COntactId
+	
+	public CW_AudienceCSVProcessHelper(Id StatusID){
+		
+		CsvStatus= [SELECT ID ,Marketing_Campaign__c,Schedule__c,Reward_Catalogue__c,Reward_Catalogue__r.Name FROM CW_CSV_Import_Status__c WHERE Id=:StatusID];
```

Comment:
Problem: Unbound SOQL query. Reason: a SOQL query can return up to 50,000 records. If this number is exceeded a runtime exception is thrown. Suggestion: cap the max number of records returned adding a LIMIT clause to the query.